ding/utils/file_helper.py

```diff
@@ -295,6 +297,32 @@ def read_file(path: str, fs_type: Union[None, str] = None, use_lock: bool = Fals
     return data
 
 
+def torch_save(data: object, path: str) -> None:
+    r"""
+    Overview:
+        torch save data to file of path, and will wait if disk is full.
+    Arguments:
+        - data (:obj:`object`): The data to save
+        - path (:obj:`str`): The path of file to save to
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # save the data to the temporary directory
+        tmp_path = f"{tmpdir}/data.pt"
+        torch.save(data, tmp_path)
+
+        # move the file to the desired path
+        while True:
+            try:
+                shutil.move(tmp_path, path)
+                break  # if successful, break out of the loop
+            except OSError as e:
+                if e.errno == 28:  # "No space left on device" error
+                    print("Disk full, waiting for space to be released...")
+                    time.sleep(60)  # wait for a minute and try again
+                else:
+                    raise e  # re-raise any other IOError exceptions
+
+
 def save_file(path: str, data: object, fs_type: Union[None, str] = None, use_lock: bool = False) -> None:
```

Comment:
add a argument to control wait or ignore disk full error

Reply:
fixed.